cdws_api/views.py

```diff
@@ -360,6 +360,30 @@ def calculate_counts(self, request, pk=None):
             data={'message': 'Calculation done.'},
             status=status.HTTP_200_OK)
 
+    @detail_route(methods=['post'],
+                  permission_classes=[IsAuthenticatedOrReadOnly])
+    def update_metrics(self, request, pk=None):
+        if 'metrics' in request.DATA and request.DATA['metrics'] != '':
+            if type(request.DATA['metrics']) is not dict:
+                return Response(
+                    status=status.HTTP_400_BAD_REQUEST,
+                    data={'message': 'Invalid metrics format'})
+            try:
+                launch = Launch.objects.get(id=pk)
+                params = launch.get_parameters()
+                params['metrics'] = request.DATA['metrics']
+                launch.set_parameters(params)
+                launch.save()
+                return Response(status=status.HTTP_200_OK,
+                                data=LaunchSerializer(launch).data)
+            except Launch.DoesNotExist:
+                return Response(
+                    data={'message':
+                          'Launch with id={} does not exist'.format(pk)},
+                    status=status.HTTP_404_NOT_FOUND)
+        return Response(status=status.HTTP_400_BAD_REQUEST,
+                        data={'message': 'No metrics in post request'})
```

Comment:
Let's add actual request.DATA to response for user can fix it problem by himself